bin/fastq_dir_to_samplesheet.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import glob
+import argparse
+
+def parse_args(args=None):
+    Description = "Generate nf-core/rnaseq samplesheet from a directory of FastQ files."
+    Epilog = "Example usage: python fastq_dir_to_samplesheet.py <FASTQ_DIR> <SAMPLESHEET_FILE>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FASTQ_DIR", help="Folder containing raw FastQ files.")
+    parser.add_argument("SAMPLESHEET_FILE", help="Output samplesheet file.")
+    parser.add_argument('-st', '--strandedness', type=str, dest="STRANDEDNESS", default='unstranded', help="Value for 'strandedness' in samplesheet. Must be one of 'unstranded', 'forward', 'reverse'.")
+    parser.add_argument('-r1', '--read1_extension', type=str, dest="READ1_EXTENSION", default='_R1_001.fastq.gz', help="File extension for read 1.")
+    parser.add_argument('-r2', '--read2_extension', type=str, dest="READ2_EXTENSION", default='_R2_001.fastq.gz', help="File extension for read 2.")
+    parser.add_argument('-se', '--single_end', dest="SINGLE_END", action='store_true', help="Single-end information will be auto-detected but this option forces paired-end FastQ files to be treated as single-end so only read 1 information is included in the samplesheet.")
+    parser.add_argument('-sn', '--sanitise_name', dest="SANITISE_NAME", action='store_true', help="Whether to further sanitise FastQ file name to get sample id. Used in conjunction with --sanitise_name_delimiter and --sanitise_name_index.")
+    parser.add_argument('-sd', '--sanitise_name_delimiter', type=str, dest="SANITISE_NAME_DELIMITER", default='_', help="Delimiter to use to sanitise sample name.")
+    parser.add_argument('-si', '--sanitise_name_index', type=int, dest="SANITISE_NAME_INDEX", default=1, help="After splitting FastQ file name by --sanitise_name_delimiter all elements before this index (1-based) will be joined to create final sample name.")
+    return parser.parse_args(args)
+
+def fastq_dir_to_samplesheet(fastq_dir, samplesheet_file, strandedness='unstranded', read1_extension='_R1_001.fastq.gz', read2_extension='_R2_001.fastq.gz', single_end=False, sanitise_name=False, sanitise_name_delimiter='_', sanitise_name_index=1):
+
+    ## Get read 1 files
+    read_dict = {}
+    read1_files = glob.glob(os.path.join(fastq_dir, f'*{read1_extension}'), recursive=False)
```

Comment:
```suggestion
    parser.add_argument('-r1', '--read1_pattern', type=str, dest="READ1_EXTENSION", default='*_R1_001.fastq.gz', help="File extension for read 1.")
    parser.add_argument('-r2', '--read2_pattern', type=str, dest="READ2_EXTENSION", default='*_R2_001.fastq.gz', help="File extension for read 2.")
    parser.add_argument('-se', '--single_end', dest="SINGLE_END", action='store_true', help="Single-end information will be auto-detected but this option forces paired-end FastQ files to be treated as single-end so only read 1 information is included in the samplesheet.")
    parser.add_argument('-sn', '--sanitise_name', dest="SANITISE_NAME", action='store_true', help="Whether to further sanitise FastQ file name to get sample id. Used in conjunction with --sanitise_name_delimiter and --sanitise_name_index.")
    parser.add_argument('-sd', '--sanitise_name_delimiter', type=str, dest="SANITISE_NAME_DELIMITER", default='_', help="Delimiter to use to sanitise sample name.")
    parser.add_argument('-si', '--sanitise_name_index', type=int, dest="SANITISE_NAME_INDEX", default=1, help="After splitting FastQ file name by --sanitise_name_delimiter all elements before this index (1-based) will be joined to create final sample name.")
    return parser.parse_args(args)

def fastq_dir_to_samplesheet(fastq_dir, samplesheet_file, strandedness='unstranded', read1_extension='_R1_001.fastq.gz', read2_extension='_R2_001.fastq.gz', single_end=False, sanitise_name=False, sanitise_name_delimiter='_', sanitise_name_index=1):

    ## Get read 1 files
    read_dict = {}
    read1_files = glob.glob(os.path.join(fastq_dir, f'{read1_extension}'), recursive=False)
```

Reply:
@drpatelh, actually, this was all I meant ;) 
(+ maybe updating the descriptions) 

But we can do that later if it's getting too much for this PR.

Reply:
Cool. Yep. Be nice to do in another PR if poss please :)